node-js-basics-101/node_modules/asynquence-contrib/contrib.src.js

```diff
@@ -0,0 +1,2360 @@
+				var args = ARRAY_SLICE.call(arguments);
+				reject.call(ø,args.length > 1 ? args : args[0]);
+			});
+		});
+	};
+});
+// "try"
+ASQ.extend("try",function $$extend(api,internals){
+	return function $$try() {
+		if (internals("seq_error") || internals("seq_aborted") ||
+			arguments.length === 0
+		) {
+			return api;
+		}
+
+		var fns = ARRAY_SLICE.call(arguments)
+		.map(function $$map(fn){
+			return function $$then(mainDone) {
+				var main_args = ARRAY_SLICE.call(arguments),
+					sq = ASQ.apply(ø,main_args.slice(1))
+				;
+
+				sq
+				.then(function $$inner$then(){
+					fn.apply(ø,arguments);
+				})
+				.val(function $$val(){
+					mainDone.apply(ø,arguments);
+				})
+				.or(function $$inner$or(){
+					var msgs = ASQ.messages.apply(ø,arguments);
+					// failed, so map error(s) as `catch`
+					mainDone({
+						"catch": msgs.length > 1 ? msgs : msgs[0]
+					});
+				});
+			};
+		});
+
+		api.then.apply(ø,fns);
+
+		return api;
+	};
+});
+// "until"
+ASQ.extend("until",function $$extend(api,internals){
+	return function $$until() {
+		if (internals("seq_error") || internals("seq_aborted") ||
+			arguments.length === 0
+		) {
+			return api;
+		}
+
+		var fns = ARRAY_SLICE.call(arguments)
+		.map(function $$map(fn){
+			return function $$then(mainDone) {
+				var main_args = ARRAY_SLICE.call(arguments),
+					sq = ASQ.apply(ø,main_args.slice(1))
+				;
+
+				sq
+				.then(function $$inner$then(){
+					var args = ARRAY_SLICE.call(arguments);
+					args[0]["break"] = function $$break(){
+						mainDone.fail.apply(ø,arguments);
+						sq.abort();
+					};
+
+					fn.apply(ø,args);
+				})
+				.val(function $$val(){
+					mainDone.apply(ø,arguments);
+				})
+				.or(function $$inner$or(){
+					// failed, retry
+					$$then.apply(ø,main_args);
+				});
+			};
+		});
+
+		api.then.apply(ø,fns);
+
+		return api;
+	};
+});
+// "waterfall"
+ASQ.extend("waterfall",function $$extend(api,internals){
+	return function $$waterfall() {
+		if (internals("seq_error") || internals("seq_aborted") ||
+			arguments.length === 0
+		) {
+			return api;
+		}
+
+		var fns = ARRAY_SLICE.call(arguments);
+
+		api.then(function $$then(done){
+			var msgs = ASQ.messages(),
+				sq = ASQ.apply(ø,ARRAY_SLICE.call(arguments,1))
+			;
+
+			fns.forEach(function $$each(fn){
+				sq.then(fn)
+				.val(function $$val(){
+					var args = ASQ.messages.apply(ø,arguments);
+					msgs.push(args.length > 1 ? args : args[0]);
+					return msgs;
+				});
+			});
+
+			sq.pipe(done);
+		});
+
+		return api;
+	};
+});
+// "wrap"
+"use strict";
+
+ASQ.wrap = function $$wrap(fn, opts) {
+	function checkThis(t, o) {
+		return !t || typeof window != "undefined" && t === window || typeof global != "undefined" && t === global ? o : t;
+	}
+
+	function paramSpread(gen) {
+		return regeneratorRuntime.mark(function paramSpread(token) {
+			return regeneratorRuntime.wrap(function paramSpread$(context$3$0) {
+				while (1) switch (context$3$0.prev = context$3$0.next) {
```

Comment:
__[jshint]__
* Expected a conditional expression and instead saw an assignment.

<!-- Comment by [SideCI](https://sideci.com) -->